kornia/geometry/nerf/nerf_solver.py

```diff
@@ -0,0 +1,154 @@
+import math
+from typing import Optional, Union
+
+import torch
+import torch.nn.functional as F
+import torch.optim as optim
+from torch import nn
+
+from kornia.geometry.camera import PinholeCamera
+from kornia.geometry.nerf.data_utils import Images, ImageTensors, RayDataset, instantiate_ray_dataloader
+from kornia.geometry.nerf.nerf_model import NerfModel
+from kornia.geometry.nerf.types import Device
+
+"""
+General design for Nerf computation:
+===================================
+Training one epoch:
+------------------
+- Create an object of class RayDataset
+- Initialize ray dataset with group of images on disk, and number of rays to randomly sample
+- Initialize a data loader with batch size info
+- Iterate over data loader
+-- Reset optimizer
+-- Run ray batch through Nerf model
+-- Find loss
+-- Back propagate loss
+-- Optimizer step
+"""
+
+
+class NerfSolver:
+    def __init__(self, device: Device) -> None:
```

Comment:
```suggestion
    def __init__(self, device: Device, config: NerfSolverConfig) -> None:
```
so that you avoid `init_training` functions which has a confusing name because doesn't start any training just setup the class

Reply:
I'm not sure I understand this suggestion. Do you recommend to have a `NerfSolverConfig` class with all training parameters that will be initialized by some convenient function, and be sent over to NerfSolver.__init__() ?